linkerd/app/outbound/src/http/logical/policy/route.rs

```diff
@@ -117,20 +116,37 @@ where
                 .push(MatchedBackend::layer(metrics.backend.clone()))
                 .lift_new_with_target()
                 .push(NewDistribute::layer())
+                .check_new::<Self>()
+                .check_new_service::<Self, http::Request<linkerd_app_core::proxy::http::BoxBody>>()
                 // The router does not take the backend's availability into
                 // consideration, so we must eagerly fail requests to prevent
                 // leaking tasks onto the runtime.
                 .push_on_service(svc::LoadShed::layer())
                 .push(filters::NewApplyFilters::<Self, _, _>::layer())
-                .push(retry::NewHttpRetry::layer(metrics.retry.clone()))
+                .push({
+                    let mk_extract = |rt: &Self| {
+                        let Route {
+                            parent_ref,
+                            route_ref,
+                            ..
+                        } = &rt.params;
+                        retry::RetryLabelExtract(parent_ref.clone(), route_ref.clone())
+                    };
+                    let metrics = metrics.retry.clone();
+                    retry::NewHttpRetry::layer_via_mk(mk_extract, metrics)
+                })
+                .check_new_service::<Self, http::Request<linkerd_app_core::proxy::http::BoxBody>>()
+                .check_new::<Self>()
                 // Set request extensions based on the route configuration
                 // AND/OR headers
                 .push(extensions::NewSetExtensions::layer())
                 .push(metrics::layer(&metrics.requests))
+                .check_new::<Self>()
+                .check_new_service::<Self, http::Request<linkerd_app_core::proxy::http::BoxBody>>()
```

Comment:
:point_up: this is mentioned in the pr description. i am happy to walk these additions back; they made my own work easier, and it seemed like an easy gift to offer future engineers tweaking our policy stack. :gift:

Reply:
I'm more than fine with these, the only real concern would be compile time impact but I don't expect these to be a burden here.

Reply:
We do tend to prune tehse down to avoid noise, but it's not really a problem to leave them in. If we do leave them in, though, I'd strongly prefer that we rely on the reexports to reduce noise:
```suggestion
                .check_new::<Self>()
                .check_new_service::<Self, http::Request<http::BoxBody>>()
```

Reply:
db00264b tweaks these to refer to the reexports. thanks for pointing that out, i hadn't realized that `http` in this scope is pointed to `linkerd_app_core::proxy::http`. :bulb:

this looks much better.